src/units/attack_type.hpp

```diff
@@ -129,6 +129,11 @@ class attack_type : public std::enable_shared_from_this<attack_type>
 	 * @param special_tags If true, match @a special against the tag name of special tags.
 	 */
 	bool has_special_or_ability(const std::string& special, bool special_id=true, bool special_tags=true) const;
+	/**
+	 * Returns true if this is a dummy attack_type, for example the placeholder that the unit_attack dialog
+	 * uses when a defender has to weapon for a given range.
+	 */
+	bool attack_empty() const {if(id().empty() && name().empty() && type().empty() && range().empty()){ return true; } return false;}
```

Comment:
No need for the if statement, `id().empty() && name().empty() && type().empty() && range().empty()` is already boolean.

Reply:
It still needs a semicolon at the end of the return statement

Reply:
fixed